client/src/hooks/useLogin.ts

```diff
@@ -0,0 +1,28 @@
+import { useAuth0 } from '@auth0/auth0-react';
+
+const serverUrl = process.env.NEXT_PUBLIC_SERVER_URL || 'http://localhost:5000';
+
+// TODO: figure out how to create a proper separation between auth0 and the dev
+// login.  How about creating two providers and conditionally wrapping the rest
+// of the site in one or the other?
+const isDev = process.env.NEXT_PUBLIC_ENVIRONMENT === 'development';
+
+export const useLogin = (): {
+  isAuthenticated: boolean;
+  login: () => Promise<void>;
+} => {
+  const { isAuthenticated, getAccessTokenSilently } = useAuth0();
+
+  const login = async () => {
+    const token = isDev ? 'fake-token' : await getAccessTokenSilently();
+
+    await fetch(`${serverUrl}/login`, {
```

Comment:
I am this getting error, after `npm run both` finish
![image](https://user-images.githubusercontent.com/88248797/182834338-15bbe580-2539-4362-928f-80d60720b3c0.png)

closing it and clicking login show same error

Reply:
Is this on GitPod?  I'll take a look!

Reply:
I am wondering if you open port `3000` in your Gitpod, does the page load?

It doesn't for me after multiple workspaces and `npm run db:reset`

Reply:
Just looking now.  I realised a few more things were broken after I fixed the issue you spotted :cold_sweat:

Reply:
this keep giving, I think there is a memory leak issue, if someone toggle between dashboard pages, the site gets slower and slower, until Gitpod crashed for me

Reply:
Nice catch. Could you replicate those steps on the main branch?  It would be good to know if this is a new bug or not.

Reply:
Sure, I will try it in the main repo and update you

Reply:
in the main repo, the snapshots start at 30-33 MB and doesn't change when toggle between them. 

In this pr, the pages start at 27-31 MB and change to 31-34 with toggle but stops there, I guess from caching. Editing increase the snapshot by 6mb once, and doesn't increase it more. 

I think that issue was a fluke from my device.

Reply:
btw what are we doing to have 30-33 MB for a page with 5 columns, that's for another issue maybe after mvp